### ISSUE TITLE
(BLE Server) sendNotification method fail callback not response 

### ISSUE THREAD

**Issue by WanDa1993** [NordicSemiconductor/Android-BLE-Library#474]
In the app the GATT Server is used. The app tries to send notifications from a characteristic for which the client hasn't enabled notifications.
The library does not report neither success not failure despite the data not being sent.

[Server]
```java
            sendNotification(mNotifyCharacteristic, bytes, 0, bytes.length)
                    .before(new BeforeCallback() {
                        @Override
                        public void onRequestStarted(@NonNull BluetoothDevice device) {
                            log(Log.VERBOSE, "onRequestStarted");
                        }
                    })
                    .invalid(new InvalidRequestCallback() {
                        @Override
                        public void onInvalidRequest() {
                            log(Log.VERBOSE, "onInvalidRequest");
                        }
                    })
                    .fail(new FailCallback() {
                        @Override
                        public void onRequestFailed(@NonNull BluetoothDevice device, int status) {
                            log(Log.VERBOSE, "onRequestFailed");
                        }
                    })
                    .done(new SuccessCallback() {
                        @Override
                        public void onRequestCompleted(@NonNull BluetoothDevice device) {
                            log(Log.VERBOSE, "onRequestCompleted");
                        }
                    })
                    .then(new AfterCallback() {
                        @Override
                        public void onRequestFinished(@NonNull BluetoothDevice device) {
                            log(Log.VERBOSE, "onRequestFinished");
                        }
                    })
                    .enqueue();
```
Reason：
https://github.com/NordicSemiconductor/Android-BLE-Library/blob/7b0b88dbcac6f215c1f8e2c5762ce701eb510797/ble/src/main/java/no/nordicsemi/android/ble/BleManagerHandler.java#L997-L1025

If the value of Client Characteristic Configuration (CCC) descriptor is null, or there is no CCC descriptor, the failure is not reported to the app.

Actual behavior:
1. notifyStarted

Expected behavior：
1. notifyStarted
2. notifyInvalidRequest or notifyFail

**Comment by philips77** [comment]
Hi,
Sorry for the delay, I was busy with other projects. I've edited your question to make it more easy to understand. I hope I got it correctly.

You're right. Right now, if notifications were not enabled there is no callback, which indeed is weird. I will think what the outcome should be. It is wrong, as when you'd call `sendNotification(...).suspend()` the suspension would never finish.

Thank you for reporting the issue.
